consensus_test.go

```diff
@@ -489,17 +508,52 @@ func TestTransition_ValidateState_MoveToCommitState(t *testing.T) {
 
 // Not enough messages are sent, so ensure that destination state is RoundChangeState and that state machine jumps out of the loop.
 func TestTransition_ValidateState_MoveToRoundChangeState(t *testing.T) {
-	m := newMockPbft(t, []NodeID{"A", "B", "C", "D", "E", "F"}, nil, "A")
-	m.setState(ValidateState)
-	m.emitMsg(createMessage(NodeID("B"), MessageReq_Commit, nil))
-	m.emitMsg(createMessage(NodeID("C"), MessageReq_Commit, nil))
+	t.Run("All the validators have the same voting powers", func(t *testing.T) {
+		m := newMockPbft(t, []NodeID{"A", "B", "C", "D", "E", "F"}, nil, "A")
+		m.setState(ValidateState)
+		m.emitMsg(createMessage(NodeID("B"), MessageReq_Commit, nil))
+		m.emitMsg(createMessage(NodeID("C"), MessageReq_Commit, nil))
 
-	m.runCycle(context.Background())
+		m.runCycle(context.Background())
 
-	m.expect(expectResult{
-		sequence:   1,
-		state:      RoundChangeState,
-		commitMsgs: 2, // Commit messages (A proposer sent commit via state machine loop, B sent commit via emit message)
+		m.expect(expectResult{
+			sequence:   1,
+			state:      RoundChangeState,
+			commitMsgs: 2, // Commit messages (A proposer sent commit via state machine loop, B sent commit via emit message)
+		})
+	})
+
+	t.Run("Validators have different voting powers (send not enough commit messages)", func(t *testing.T) {
+		m := newMockPbft(t, []NodeID{"A", "B", "C", "D", "E", "F"}, map[NodeID]uint64{"A": 10, "B": 15, "C": 5, "D": 20}, "A")
+		m.setState(ValidateState)
+		m.emitMsg(createMessage(NodeID("B"), MessageReq_Commit, nil))
+		m.emitMsg(createMessage(NodeID("C"), MessageReq_Commit, nil))
+
+		m.runCycle(context.Background())
+
+		m.expect(expectResult{
+			sequence:   1,
+			state:      RoundChangeState,
+			commitMsgs: 2, // Commit messages (A proposer sent commit via state machine loop, B sent commit via emit message)
```

Comment:
We should add another condition here to check the aggregated voting power as well.

Reply:
Agreed.

Reply:
Fixed https://github.com/0xPolygon/pbft-consensus/pull/73/commits/386a435ae2576f0cc14ccebbff9c5c9faf3d8502